tg/MainScreenListOfGames/ListOfGamesViewController.swift

```diff
@@ -48,87 +45,94 @@ class ListOfGamesViewController: UIViewController {
     } */
-    
-    
     func receiveDataFromServer() {
-        
         guard let url = URL(string: Properties.linkForData) else {return}
 
         NetworkManager.networkManager.getDataFromServer(url, complitionHandler: { data in
-                
-                self.gameList = data.results ?? []
+            self.gameList = data.results ?? []
         })
     }
-    
     func loadLogoOfGames() {
-        
         for item in gameList {
-            
             var readyImage = UIImage(named: "dice")
-            
             if let backImage = item.backgroundImage {
                 if let data = NSData(contentsOf: NSURL(string: backImage)! as URL) {
                     readyImage = UIImage(data: data as Data)
                 }
             }
-            
             dictionaryOfLogo[item.id] = readyImage
         }
-        
         tableListOfGame.reloadData()
-        
+    }
+    func loadingAnimation() {
+        let sizeOfCircle = Properties.sizeOfLoadCircle
+        let widthOfCircleOutside = sizeOfCircle
+        let heightOfCircleOutside = widthOfCircleOutside
+        let xOfCircleOutside = tableListOfGame.frame.width / 2 - (sizeOfCircle / 2)
+        let yOfCircleOutside = tableListOfGame.frame.height / 2 - (sizeOfCircle / 2)
+        let widthOfCircleInside = widthOfCircleOutside - (sizeOfCircle / 6)
+        let heightOfCircleInside = widthOfCircleInside
+        let xOfCircleInside = sizeOfCircle / 12
+        let yOfCircleInside = xOfCircleInside
+        let widthOfMovingView = widthOfCircleOutside
+        let heightOfMovingView = widthOfMovingView / 3
+        let rangeOfMotion = widthOfMovingView - heightOfMovingView
+        
+        animatedСircleOutside = UIView(frame: CGRect(x: xOfCircleOutside, y: yOfCircleOutside, width: widthOfCircleOutside, height: heightOfCircleOutside))
+        animatedСircleOutside.clipsToBounds = true
+        animatedСircleOutside.layer.cornerRadius = widthOfCircleOutside / 2
+        tableListOfGame.addSubview(animatedСircleOutside)
+        
+        let animatedMovingView = UIView(frame: CGRect(x: 0, y: 0, width: widthOfMovingView, height: heightOfMovingView))
+        animatedMovingView.backgroundColor = .systemGray3
+        animatedСircleOutside.addSubview(animatedMovingView)
+        
+        let animatedСircleInside = UIView(frame: CGRect(x: xOfCircleInside, y: yOfCircleInside, width: widthOfCircleInside, height: heightOfCircleInside))
+        animatedСircleInside.backgroundColor = .white
+        animatedСircleInside.clipsToBounds = true
+        animatedСircleInside.layer.cornerRadius = widthOfCircleInside / 2
+        animatedСircleOutside.addSubview(animatedСircleInside)
+        
+        UIView.animate(withDuration: 0.5, delay: 0, options: [.repeat, .autoreverse]) {
+            animatedMovingView.frame.origin = CGPoint(x: 0, y: rangeOfMotion)
+        }
```

Comment:
If you will need to use this animation on other viewController you will have to copy paste code. So LoadingAnimation has to be a reusable view. Also, try to change implementation and use UIBezierPath instead of UIView